### PR TITLE
Updated isomorphic-fetch's use of global.require because it breaks in node env

### PR DESCRIPTION
**Pull request by yomexzo**
👀 @minasmart @tessalt @richgilbank

### CODE REVIEW

File: src/isomorphic-fetch.js [hunk 1]
```diff
@@ -1,9 +1,13 @@
+/* globals require */
+
 import global from './metal/global';
 import isNodeLikeEnvironment from './metal/is-node-like-environment';
 
 const fetch = global.fetch;
 
 if (!fetch && isNodeLikeEnvironment()) {
-  global.fetch = global.require('node-fetch');
+  const localRequire = require;
```
Comment (minasmart) [inline src/isomorphic-fetch.js]: Can you verify that this indirection is sufficient to pass `npm run build`? I would also annotate the reason for the indirection.

Comment (yomexzo) [inline src/isomorphic-fetch.js]: yeah. verified.
added comment too.

File: src/isomorphic-fetch.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 import global from './metal/global';
 import isNodeLikeEnvironment from './metal/is-node-like-environment';
 
 const fetch = global.fetch;
 
 if (!fetch && isNodeLikeEnvironment()) {
-  global.fetch = global.require('node-fetch');
+  /* global.require is only available in the REPL 
+   * and Babel fails when you use require() explicitly
+   */
```
Comment (minasmart) [inline src/isomorphic-fetch.js]: More than babel failing, we can assign this specifically to the plugin `babel-plugin-transform-es2015-modules-amd`

Comment (yomexzo) [inline src/isomorphic-fetch.js]: hmm... i'm wondering, is it `amd` or `commonjs`? i thought the `commonjs` versions were loaded when the library is required/used in the nodejs environment.

Comment (minasmart) [inline src/isomorphic-fetch.js]: You can repeatably cause this bug under the amd processor. The commonjs one seems fine. There's also a long outstanding bug which I couldn't find documenting it only on the amd processor.

Comment (yoadsn) [inline src/isomorphic-fetch.js]: I can confirm that on my side.

Comment (yomexzo) [inline src/isomorphic-fetch.js]: @minasmart, i get what you mean now. the problem also happens when transpiling to `umd`. It's a problem with babel6. I found situations when it choked too when transpiling scripts with `decorators`. I have updated the comment now

**Comment by yomexzo** [comment]
@m-ux, @minasmart .. kindly 👀 

this change is needed because `global.require` yields `undefined` when the library is being used in node.

i had to use an intermediate variable instead of just using `require()` because transpiling to `amd` or even `umd` with `babel6` fails when `require` is used explicitly with an error with message like `TypeError: src/isomorphic-fetch.js: Property right of AssignmentExpression expected node to be of a type ["Expression"] but instead got null`

fixes #153

**Comment by minasmart** [comment]
Can this 🚢 ?

**Pull request merged**